selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/site/CheLoginPage.java

```diff
@@ -11,66 +11,101 @@
-import org.eclipse.che.selenium.core.constant.TestTimeoutsConstants;
-import org.openqa.selenium.By;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
 import org.openqa.selenium.TimeoutException;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.PageFactory;
-import org.openqa.selenium.support.ui.ExpectedCondition;
-import org.openqa.selenium.support.ui.WebDriverWait;
 
 /** @author Dmytro Nochevnov */
 @Singleton
 public class CheLoginPage implements LoginPage {
-  private final SeleniumWebDriver seleniumWebDriver;
-  private final WebDriverWait webDriverWait;
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
 
-  protected interface LoginPageLocators {
-    String USER_NAME_FIELD = "username";
-    String PASSWORD_FIELD = "password";
-    String LOGIN_BUTTON = "login";
+  private static final String FIRST_NAME = "first name";
+  private static final String LAST_NAME = "last name";
+
+  protected interface Locators {
+    String USERNAME_INPUT_NAME = "username";
+    String PASSWORD_INPUT_NAME = "password";
+    String LOGIN_BUTTON_NAME = "login";
+    String OPEN_SHIFT_OAUTH_LINK_ID = "zocial-openshift-v3";
+
+    String ERROR_ALERT_XPATH = "//div[@class='alert alert-error']/span[@class='kc-feedback-text']";
+    String INFO_ALERT_XPATH = "//div[@class='alert alert-info']/span[@class='kc-feedback-text']";
   }
 
-  @FindBy(name = USER_NAME_FIELD)
+  @FindBy(name = USERNAME_INPUT_NAME)
   private WebElement usernameInput;
 
-  @FindBy(name = PASSWORD_FIELD)
+  @FindBy(name = PASSWORD_INPUT_NAME)
   private WebElement passwordInput;
 
-  @FindBy(name = LOGIN_BUTTON)
+  @FindBy(name = LOGIN_BUTTON_NAME)
   private WebElement loginButton;
 
+  @FindBy(id = OPEN_SHIFT_OAUTH_LINK_ID)
+  private WebElement openShiftOAuthLink;
+
+  @FindBy(xpath = Locators.INFO_ALERT_XPATH)
+  private WebElement infoAlert;
+
+  @FindBy(xpath = Locators.ERROR_ALERT_XPATH)
+  private WebElement errorAlert;
+
   @Inject
-  public CheLoginPage(SeleniumWebDriver seleniumWebDriver) {
-    this.seleniumWebDriver = seleniumWebDriver;
-    PageFactory.initElements(seleniumWebDriver, this);
+  public CheLoginPage(
+      SeleniumWebDriver seleniumWebDriver, SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
 
-    webDriverWait = new WebDriverWait(seleniumWebDriver, TestTimeoutsConstants.LOADER_TIMEOUT_SEC);
+    PageFactory.initElements(seleniumWebDriver, this);
   }
 
   @Override
   public void login(String username, String password) {
     waitOnOpen();
 
-    rewrite(usernameInput, username);
+    seleniumWebDriverHelper.setValue(usernameInput, username);
+    seleniumWebDriverHelper.setValue(passwordInput, password);
+    seleniumWebDriverHelper.waitAndClick(loginButton);
 
-    rewrite(passwordInput, password);
+    waitOnClose();
+  }
+
+  public void loginWithPredefinedUsername(String password) {
```

Comment:
I think you can get rid of "WithPredefinedUsername". In my opinion "login(String password)" looks better.

Reply:
Login form with predefined username is not the same as login WITHOUT username (which means we are leaving username untouched). Here we have completely different scenario when the username input contains PREDEFINED username which we can even not change.

Reply:
For me, this is clear from method signature. Of course, you may have your own vision.